pkg/client/client.go

```diff
@@ -968,6 +970,12 @@ func (c *Client) ValidatePrometheus(ctx context.Context, promNsName types.Namesp
 	}, WithPollInterval(10*time.Second))
 
 	if pollErr != nil {
+		s, _ := c.kclient.AppsV1().StatefulSets(promNsName.Namespace).Get(ctx, "prometheus-k8s", metav1.GetOptions{})
+		generationAttemps := s.Status.ObservedGeneration - lastGeneration
+		if generationAttemps > 5 && s.Status.ReadyReplicas != s.Status.Replicas {
```

Comment:
(from Simon's comment above)
```suggestion
		if generationAttemps > 10 && s.Status.ReadyReplicas != s.Status.Replicas {
```